crates/ruff_cli/src/args.rs

```diff
@@ -74,7 +74,8 @@ pub enum Command {
     },
 }
 
-#[derive(Clone, Debug, clap::Args)]
+// The `Parser` derive is for ruff_dev, for ruff_cli `Args` would be sufficient
+#[derive(Clone, Debug, clap::Parser)]
```

Comment:
Does this change the size of the resulting binary? If so, move behind a feature flag? Why does the existing `WrapperArgs` approach no longer work?

Reply:
`WrapperArgs` was mostly a hack around clap's limitations, i think this is a much cleaner solution.

For `ruff-0.0.277-py3-none-manylinux_2_34_x86_64.whl` from `CARGO_TARGET_DIR=target-maturin maturin build --release --strip -j 2`:
* On main: 5593346
* This branch: 5504378

Surprisingly even better than main, so i'd say the impact on binary size is in the realm of noise.